services/localfile/client/client.go

```diff
@@ -978,14 +978,15 @@ func (p *lsCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{
 }
 
 type cpCmd struct {
-	bucket    string
-	overwrite bool
-	uid       int
-	username  string
-	gid       int
-	group     string
-	mode      int
-	immutable bool
+	bucket               string
+	overwrite            bool
+	createDestinationDir bool
```

Comment:
I can't help but notice that the style of those options is usually one word long, pretty terse. Here a flag with a longer name made of multiple words is added - which to me looks a bit like a red flag.

Additionally, in Unix systems, a different command is used to create directories and it's not an option for `cp`.

I haven't worked on this repo before, so it's hard for me to tell whether this is in line with its usual style. This is where I would like to defer to the maintainers.